spaceship/lib/spaceship/connect_api/api_client.rb

```diff
@@ -199,8 +201,17 @@ def handle_response(response)
         return Spaceship::ConnectAPI::Response.new(body: response.body, status: response.status, headers: response.headers, client: self)
       end
 
-      def handle_401(response)
-        raise UnauthorizedAccessError, handle_errors(response) if response && (response.body || {})['errors']
+      def handle_4xx_errors(response)
+        case response.status.to_i
+        when 401
+          raise UnauthorizedAccessError, handle_errors(response) if response && (response.body || {})['errors']
+        when 403
+          if response.body['errors'].first['code'].eql?("FORBIDDEN.REQUIRED_AGREEMENTS_MISSING_OR_EXPIRED")
```

Comment:
I really don't like this line! Any suggestions for doing this better is more than welcome 🙏